src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +13,400 @@
+        List<FeedbackParticipantType> participantVisiblity = new ArrayList<FeedbackParticipantType>();
+        participantVisiblity.add(FeedbackParticipantType.INSTRUCTORS);
+        participantVisiblity.add(FeedbackParticipantType.RECEIVER);
+        
+        for (FeedbackParticipantType participant : participantVisiblity) {
+            isGiverNameVisible.put(participant.name(), true);
+            isRecipientNameVisible.put(participant.name(), true);
+            isResponsesVisible.put(participant.name(), true);
+        }
+        
+        generalSettings.setGiverNameVisibleFor(isGiverNameVisible);
+        generalSettings.setRecipientNameVisibleFor(isRecipientNameVisible);
+        generalSettings.setResponseVisibleFor(isResponsesVisible);
+    }
+    
+    
+    private void buildBasicFsForm(FeedbackSessionAttributes newFeedbackSession) {
+        fsForm = new FeedbackSessionsForm();
+        
+        fsForm.setFsDeleteLink(new Url(getInstructorFeedbackSessionDeleteLink(newFeedbackSession.courseId, newFeedbackSession.feedbackSessionName, "")));
+        fsForm.setCopyToLink(new Url(getFeedbackSessionEditCopyLink()));
+        
+        fsForm.setCourseIdForNewSession(newFeedbackSession.courseId);
+        
+        fsForm.setFsNameEditable(false);
+        fsForm.setFsName(newFeedbackSession.feedbackSessionName);
+        
+        fsForm.setCourseIdEditable(false);
+        fsForm.setCourses(null);
+        
+        fsForm.setEditFsButtonsVisible(true);
+        fsForm.setFeedbackSessionTypeEditable(false);
+        fsForm.setFeedbackSessionTypeOptions(null);
+
+        fsForm.setFeedbackSessionNameForSessionList(null);
+        
+        fsForm.setCoursesSelectField(null);
+        
+        fsForm.setTimezoneSelectField(getTimeZoneOptionsAsElementTags(newFeedbackSession.timeZone));
+        
+        
+        fsForm.setInstructions(sanitizeForHtml(newFeedbackSession.instructions.getValue()));
+        
+        fsForm.setFsStartDate(TimeHelper.formatDate(newFeedbackSession.startTime));
+        
+        Date date;
+        date = newFeedbackSession.startTime;
+        fsForm.setFsStartTimeOptions(getTimeOptionsAsElementTags(date));
+        
+        fsForm.setFsEndDate(TimeHelper.formatDate(newFeedbackSession.endTime));
+        
+        
+        date = newFeedbackSession.endTime;
+        fsForm.setFsEndTimeOptions(getTimeOptionsAsElementTags(date));
+        
+        fsForm.setGracePeriodOptions(getGracePeriodOptionsAsElementTags(newFeedbackSession.gracePeriod));
+        
+        fsForm.setSubmitButtonDisabled(false);
+        fsForm.setFormSubmitAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_EDIT_SAVE));
+        fsForm.setSubmitButtonText(SAVE_CHANGES);
+        fsForm.setSubmitButtonVisible(false);
+        
+        fsForm.setEditButtonTags(new ElementTag("onclick", "enableEditFS()"));
+    }
+
+    private AdditionalSettingsFormSegment buildFsFormAdditionalSettings(FeedbackSessionAttributes newFeedbackSession) {
```

Comment:
Rename AdditionSettingsFormSegment to make it clear that it's used on FeedbackSessionForm